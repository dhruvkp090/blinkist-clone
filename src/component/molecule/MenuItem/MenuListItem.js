import React from 'react';

import MenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import { Typography } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import icons from '../../../assets/icons';


const addIcon = (name) => {
    try  { 
        return icons[name].src;
    }
    catch (e) {
        return icons["Test"].src;
    }
};

const MenuListItem = ({listItems, ...props}) => {
    return (
        listItems.map((listItem) => (
            <Grid item xs={12} sm={6} md={4} key={props.unique}>
                <MenuItem
                    {...props}>
                    <ListItemIcon>
                        {addIcon(listItem.name)}
                    </ListItemIcon>
                    <Typography variant="h3" style={{paddingTop: "0.5rem"}}>
                        {listItem.name}
                    </Typography>
                </MenuItem>
            </Grid>
        ))
    );
};

export default MenuListItem;
