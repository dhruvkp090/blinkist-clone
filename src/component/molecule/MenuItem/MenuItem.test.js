import React from 'react';
import {render} from '@testing-library/react';
import MenuListItem from './MenuListItem';
import '@testing-library/jest-dom';

const listItems= [
        {
            "id": 1,
            "name": "Entrepreneurship"
        },
        {
            "id": 2,
            "name": "Science"
        },
        {
            "id": 3,
            "name": "Economics"
        },
        {
            "id": 4,
            "name": "Politics"
        },
        {
            "id": 5,
            "name": "History"
        },
        {
            "id": 6,
            "name": "Marketing & Sales"
        }
    ];

const listItemsRandom= [
        {
            "id": 1,
            "name": "Entrepreneurship"
        },
        {
            "id": 2,
            "name": "Random name"
        }
    ];

it('Snapshot menu list', () => {
    const { asFragment } = render(<MenuListItem listItems={listItems} />);

    expect(asFragment()).toMatchSnapshot();
});

it('Snapshot menu list with random names', () => {
    const { asFragment } = render(<MenuListItem listItems={listItemsRandom} />);

    expect(asFragment()).toMatchSnapshot();
});
