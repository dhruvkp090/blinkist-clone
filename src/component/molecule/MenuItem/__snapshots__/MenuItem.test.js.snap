// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot menu list 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-rocket fa-w-16 "
          data-icon="rocket"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.12019,19.09375c-1.18945-5.53125-6.65819-11-12.207-12.1875C460.716,0,435.507,0,410.40747,0,307.17523,0,245.26909,55.20312,199.05238,128H94.83772c-16.34763.01562-35.55658,11.875-42.88664,26.48438L2.51562,253.29688A28.4,28.4,0,0,0,0,264a24.00867,24.00867,0,0,0,24.00582,24H127.81618l-22.47457,22.46875c-11.36521,11.36133-12.99607,32.25781,0,45.25L156.24582,406.625c11.15623,11.1875,32.15619,13.15625,45.27726,0l22.47457-22.46875V488a24.00867,24.00867,0,0,0,24.00581,24,28.55934,28.55934,0,0,0,10.707-2.51562l98.72834-49.39063c14.62888-7.29687,26.50776-26.5,26.50776-42.85937V312.79688c72.59753-46.3125,128.03493-108.40626,128.03493-211.09376C512.07526,76.5,512.07526,51.29688,505.12019,19.09375ZM384.04033,168A40,40,0,1,1,424.05,128,40.02322,40.02322,0,0,1,384.04033,168Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
        style="padding-top: 0.5rem;"
      >
        Entrepreneurship
      </h3>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-flask fa-w-14 "
          data-icon="flask"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M437.2 403.5L320 215V64h8c13.3 0 24-10.7 24-24V24c0-13.3-10.7-24-24-24H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h8v151L10.8 403.5C-18.5 450.6 15.3 512 70.9 512h306.2c55.7 0 89.4-61.5 60.1-108.5zM137.9 320l48.2-77.6c3.7-5.2 5.8-11.6 5.8-18.4V64h64v160c0 6.9 2.2 13.2 5.8 18.4l48.2 77.6h-172z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
        style="padding-top: 0.5rem;"
      >
        Science
      </h3>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-rupee-sign fa-w-10 "
          data-icon="rupee-sign"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M308 96c6.627 0 12-5.373 12-12V44c0-6.627-5.373-12-12-12H12C5.373 32 0 37.373 0 44v44.748c0 6.627 5.373 12 12 12h85.28c27.308 0 48.261 9.958 60.97 27.252H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h158.757c-6.217 36.086-32.961 58.632-74.757 58.632H12c-6.627 0-12 5.373-12 12v53.012c0 3.349 1.4 6.546 3.861 8.818l165.052 152.356a12.001 12.001 0 0 0 8.139 3.182h82.562c10.924 0 16.166-13.408 8.139-20.818L116.871 319.906c76.499-2.34 131.144-53.395 138.318-127.906H308c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-58.69c-3.486-11.541-8.28-22.246-14.252-32H308z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
        style="padding-top: 0.5rem;"
      >
        Economics
      </h3>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-landmark fa-w-16 "
          data-icon="landmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M501.62 92.11L267.24 2.04a31.958 31.958 0 0 0-22.47 0L10.38 92.11A16.001 16.001 0 0 0 0 107.09V144c0 8.84 7.16 16 16 16h480c8.84 0 16-7.16 16-16v-36.91c0-6.67-4.14-12.64-10.38-14.98zM64 192v160H48c-8.84 0-16 7.16-16 16v48h448v-48c0-8.84-7.16-16-16-16h-16V192h-64v160h-96V192h-64v160h-96V192H64zm432 256H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h480c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
        style="padding-top: 0.5rem;"
      >
        Politics
      </h3>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-monument fa-w-12 "
          data-icon="monument"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M368 448H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h352c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zm-78.86-347.26a31.97 31.97 0 0 0-9.21-19.44L203.31 4.69c-6.25-6.25-16.38-6.25-22.63 0l-76.6 76.61a31.97 31.97 0 0 0-9.21 19.44L64 416h256l-30.86-315.26zM240 307.2c0 6.4-6.4 12.8-12.8 12.8h-70.4c-6.4 0-12.8-6.4-12.8-12.8v-38.4c0-6.4 6.4-12.8 12.8-12.8h70.4c6.4 0 12.8 6.4 12.8 12.8v38.4z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
        style="padding-top: 0.5rem;"
      >
        History
      </h3>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chart-line fa-w-16 "
          data-icon="chart-line"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
        style="padding-top: 0.5rem;"
      >
        Marketing & Sales
      </h3>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
</DocumentFragment>
`;

exports[`Snapshot menu list with random names 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-rocket fa-w-16 "
          data-icon="rocket"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.12019,19.09375c-1.18945-5.53125-6.65819-11-12.207-12.1875C460.716,0,435.507,0,410.40747,0,307.17523,0,245.26909,55.20312,199.05238,128H94.83772c-16.34763.01562-35.55658,11.875-42.88664,26.48438L2.51562,253.29688A28.4,28.4,0,0,0,0,264a24.00867,24.00867,0,0,0,24.00582,24H127.81618l-22.47457,22.46875c-11.36521,11.36133-12.99607,32.25781,0,45.25L156.24582,406.625c11.15623,11.1875,32.15619,13.15625,45.27726,0l22.47457-22.46875V488a24.00867,24.00867,0,0,0,24.00581,24,28.55934,28.55934,0,0,0,10.707-2.51562l98.72834-49.39063c14.62888-7.29687,26.50776-26.5,26.50776-42.85937V312.79688c72.59753-46.3125,128.03493-108.40626,128.03493-211.09376C512.07526,76.5,512.07526,51.29688,505.12019,19.09375ZM384.04033,168A40,40,0,1,1,424.05,128,40.02322,40.02322,0,0,1,384.04033,168Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
        style="padding-top: 0.5rem;"
      >
        Entrepreneurship
      </h3>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-flask fa-w-14 "
          data-icon="flask"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M437.2 403.5L320 215V64h8c13.3 0 24-10.7 24-24V24c0-13.3-10.7-24-24-24H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h8v151L10.8 403.5C-18.5 450.6 15.3 512 70.9 512h306.2c55.7 0 89.4-61.5 60.1-108.5zM137.9 320l48.2-77.6c3.7-5.2 5.8-11.6 5.8-18.4V64h64v160c0 6.9 2.2 13.2 5.8 18.4l48.2 77.6h-172z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
        style="padding-top: 0.5rem;"
      >
        Random name
      </h3>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
</DocumentFragment>
`;
